tests/scripts/thread-cert/Cert_5_2_05_AddressQuery.py

```diff
@@ -219,17 +264,174 @@ def test(self):
                 flag2002 += 1
             else:
                 raise "Error: Address is unexpected."
-            self.assertTrue(self.nodes[ED1].ping(global_address))
+            self.assertTrue(self.nodes[MED].ping(global_address))
 
             # Wait for sniffer collecting packets
             self.simulator.go(1)
 
-            reed_messages = self.simulator.get_messages_sent_by(DUT_REED)
-            msg = reed_messages.next_coap_message('0.02', '/a/an')
-            command.check_address_notification(msg, self.nodes[DUT_REED], self.nodes[LEADER])
+    def verify(self, pv):
+        pkts = pv.pkts
+        pv.summary.show()
+
+        LEADER_RLOC = pv.vars['LEADER_RLOC']
+        LEADER_MLEID = pv.vars['LEADER_MLEID']
+        ROUTER_1 = pv.vars['ROUTER_1']
+        REED = pv.vars['REED']
+        REED_MLEID = pv.vars['REED_MLEID']
+        REED_RLOC = pv.vars['REED_RLOC']
+        MED = pv.vars['MED']
+        MED_MLEID = pv.vars['MED_MLEID']
+        MM = pv.vars['MM_PORT']
+        REED2001 = ''
+        REED2002 = ''
+        MED2001 = ''
+        MED2002 = ''
+
+        for addr in pv.vars['REED_IPADDRS']:
+            if addr.startswith(Bytes('2001')):
+                REED2001 = addr
+            if addr.startswith(Bytes('2002')):
+                REED2002 = addr
+
+        for addr in pv.vars['MED_IPADDRS']:
+            if addr.startswith(Bytes('2001')):
+                MED2001 = addr
+            if addr.startswith(Bytes('2002')):
+                MED2002 = addr
+
+        # Step 3: Verify topology is formed correctly except REED.
+
+        pv.verify_attached('ROUTER_2')
+        for i in range(1, 16):
+            with pkts.save_index():
+                if i == 2:
+                    continue
+                pv.verify_attached('ROUTER_%d' % i)
```

Comment:
Is it necessary to explicitly verify `ROUTER_2`?

Reply:
will add ROUTER_2  in the loop